pi5/pinctrl_utils.go

```diff
@@ -125,6 +124,29 @@ func getRegAddr(childNodePath string, numPAddrCells uint32) (uint64, error) {
 	return physAddr, err
 }
 
+// read in 'numCells' 32 bit chunks from rangeByteContents, the bytestream outputted from reading the file '/ranges'. Convert bytes into their uint64 value
+func parseBytstreamCells(numCells uint32, rangeByteContents []byte) uint64 {
+	var parsedValue uint64
+
+	switch numCells {
+
+	// reading in 32 bits. regardless we must convert to a 64 bit address so we add a bunch of 0s to the beginning.
+	case 1:
+		parsedValue = uint64(binary.BigEndian.Uint32(rangeByteContents[:(4 * numCells)]))
```

Comment:
is there anyway this could error?

Reply:
I can add some error checking related to the num bytes from a file not lining up with the inputted numAddress / numSize cells. However, this should never happen. Those numbers are taken directly from the device-tree itself. what do you think?

Reply:
yeah lets add error checking for that, just so if something unexpected happens it is easier to debug in the future